vyper/ast/parse.py

```diff
@@ -238,25 +241,50 @@ def visit_For(self, node):
                 node.col_offset,
             )
 
+        self.generic_visit(node)
+
         try:
-            annotation = python_ast.parse(raw_annotation, mode="eval")
-            # annotate with token and source code information. `first_token`
-            # and `last_token` attributes are accessed in `generic_visit`.
-            tokens = asttokens.ASTTokens(raw_annotation)
-            tokens.mark_tokens(annotation)
+            annotation_str = tokenize.untokenize(annotation_tokens).strip(string.whitespace + "\\")
+            annotation = python_ast.parse(annotation_str)
         except SyntaxError as e:
             raise SyntaxException(
                 "invalid type annotation", self._source_code, node.lineno, node.col_offset
             ) from e
 
-        assert isinstance(annotation, python_ast.Expression)
-        annotation = annotation.body
-
-        old_target = node.target
-        new_target = python_ast.AnnAssign(target=old_target, annotation=annotation, simple=1)
-        node.target = new_target
+        annotation = annotation.body[0]
+        og_target = node.target
+
+        # annotate with token and source code information. `first_token`
+        # and `last_token` attributes are accessed in `generic_visit`.
+        tokens = asttokens.ASTTokens(annotation_str)
+        tokens.mark_tokens(annotation)
+
+        # decrease line offset by 1 because annotation is on the same line as `For` node
+        # but the spliced expression also starts at line 1
+        adjustment = og_target.first_token.start[0] - 1, og_target.first_token.start[1]
+
+        def _add_pair(x, y):
+            return x[0] + y[0], x[1] + y[1]
+
+        for n in python_ast.walk(annotation):
+            # adjust all offsets
+            if hasattr(n, "first_token"):
+                n.first_token = n.first_token._replace(
+                    start=_add_pair(n.first_token.start, adjustment),
+                    end=_add_pair(n.first_token.end, adjustment),
+                    startpos=n.first_token.startpos + og_target.first_token.startpos,
+                    endpos=n.first_token.startpos + og_target.first_token.startpos,
```

Comment:
```suggestion
                    endpos=n.first_token.endpos + og_target.first_token.startpos,
```
Should this be `endpos`?

Reply:
False alarm.